src/reconcile/rec_row.c

```diff
@@ -919,7 +919,8 @@ __wt_rec_row_leaf(
                      */
                     if (!F_ISSET(session, WT_SESSION_NO_DATA_HANDLES)) {
                         WT_ERR(__wt_hs_cursor_open(session));
-                        WT_ERR(__wt_hs_delete_key_from_ts(session, btree->id, tmpkey, WT_TS_NONE));
+                        WT_ERR(
+                          __wt_hs_reinsert_key_from_ts(session, btree->id, tmpkey, WT_TS_NONE));
```

Comment:
I think here may be there is a problem. In this case, we have decided to delete the key from the data store so we want everything of that key in the history store also gone. I don't think we need to reinsert anything in this case.

Reply:
May be we should give __wt_hs_reinsert_key_from_ts a bool argument to indicate whether we need to reinsert or not.

Reply:
Yeah okay I'll add that in quickly.

Reply:
50ee4d7

Reply:
A small suggestion, shall we also rename __wt_hs_reinsert_key_from_ts to __wt_hs_fixup_key_from_ts or some other name?

Reply:
Yeah for correctness reasons that is better

Reply:
2b496ad